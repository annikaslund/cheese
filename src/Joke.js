import React, { Component } from 'react';

class Joke extends Component {
    // if count total has changed, will only re-render that Joke
    shouldComponentUpdate(nextProps, nextState) {
        return this.props.total !== nextProps.total
    }

    // renders a joke containing joke text, buttons for upvoting and downvoting, and the net total of votes.
    render(){
        return (
            <div className="Joke">
                {this.props.jokeText}
                <button onClick={this.props.handleUpClick}>Up</button>
                {this.props.total}
                <button onClick={this.props.handleDownClick}>Down</button>
            </div>
        );
    }
}

export default Joke;