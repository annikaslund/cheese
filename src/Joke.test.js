import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Joke from "./Joke";

// smoke test
it("renders without crashing", function() {
    mount(<Joke />);
});
  
  // snapshot test
it("matches snapshot", function() {
    let wrapper = mount(<Joke />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});