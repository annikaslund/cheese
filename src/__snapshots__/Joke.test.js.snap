// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Joke>
  <div
    className="Joke"
  >
    <button>
      Up
    </button>
    <button>
      Down
    </button>
  </div>
</Joke>
`;
